### PR TITLE
[17.0][MIG] edi_voxel_account_invoice_oca: Migration to 17.0

### PR DESCRIPTION
**Pull request by FernandoRomera**

### CODE REVIEW

**Comment by FernandoRomera** [comment]
@pedrobaeza 
This is the first time this has happened to me, I have to downgrade this PR to Beta?.

Run manifestoo -d . check-dev-status --default-dev-status=Beta
edi_voxel_account_invoice_oca (Production/Stable) depends on stock_picking_invoice_link (Beta)
Error: Process completed with exit code 1.

**Comment by pedrobaeza** [comment]
I think you should promote instead `stock_picking_invoice_link` to `Mature`. The problem is that this check was not done in previous versions, and the default one for all the modules is `Beta`.

**Comment by FernandoRomera** [comment]
Depend:

- [x] <https://github.com/OCA/stock-logistics-workflow/pull/1667>

**Comment by FernandoRomera** [comment]
@pedrobaeza 
Done, can you help me to merge it <https://github.com/OCA/stock-logistics-workflow/pull/1667>?

**Comment by FernandoRomera** [comment]
@pedrobaeza 
Thank you for all your help.
I made a mistake in the previous PR <https://github.com/OCA/edi/pull/1020>, pending merging.

**Review by AntoniRomera — APPROVED** [review]
LGTM

**Comment by FernandoRomera** [comment]
@pedrobaeza 
Done, can you review and merge?.

File: edi_voxel_account_invoice_oca/tests/test_voxel_account_invoice.py [hunk 1]
```diff
@@ -238,7 +251,7 @@ def _get_products_data(self):
                     "UP": "750.0",
                     "Total": "1500.0",
                 },
-                "taxes": [],
+                "taxes": [{"Type": False, "Rate": "15.0"}],
```
Comment (pedrobaeza) [inline edi_voxel_account_invoice_oca/tests/test_voxel_account_invoice.py]: Why this change and the following one?

Comment (FernandoRomera) [reply edi_voxel_account_invoice_oca/tests/test_voxel_account_invoice.py]: Simply compare the result with the expected result and match the values.
It seems to assign a default value to the tax_ids field in the invoice lines if it does not have one and therefore all values change. I can leave it like this or set tax_ids: False, as you prefer.

Comment (pedrobaeza) [reply edi_voxel_account_invoice_oca/tests/test_voxel_account_invoice.py]: Yes, we should fix it and don't leave this to the "default behavior in some cases".

Comment (FernandoRomera) [reply edi_voxel_account_invoice_oca/tests/test_voxel_account_invoice.py]: Done, set to 15% tax for lines without.

Comment (pedrobaeza) [reply edi_voxel_account_invoice_oca/tests/test_voxel_account_invoice.py]: On contrary, what I say is to have it empty.

File: edi_voxel_account_invoice_oca/tests/test_voxel_account_invoice.py [hunk 2]
```diff
@@ -109,6 +110,7 @@ def setUpClass(cls):
                             "product_id": product_1.id,
                             "quantity": 2,
                             "price_unit": 750,
+                            "tax_ids": [(6, 0, (tax_15).ids)],
```
Comment (pedrobaeza) [inline edi_voxel_account_invoice_oca/tests/test_voxel_account_invoice.py]:
```suggestion
                            "tax_ids": False,
```

and the other one

Comment (FernandoRomera) [reply edi_voxel_account_invoice_oca/tests/test_voxel_account_invoice.py]: @pedrobaeza 
Ok, done.

File: edi_voxel_account_invoice_oca/tests/test_voxel_account_invoice.py [hunk 3]
```diff
@@ -218,7 +222,19 @@ def _get_client_data(self):
         }
 
     def _get_customers_data(self):
-        return []
+        return [
+            {
+                "SupplierClientID": "C01",
```
Comment (pedrobaeza) [inline edi_voxel_account_invoice_oca/tests/test_voxel_account_invoice.py]: Why is this changing as well?

File: edi_voxel_account_invoice_oca/tests/test_voxel_account_invoice.py [hunk 4]
```diff
@@ -281,13 +297,9 @@ def _get_products_data(self):
 
     def _get_taxes_data(self):
         return [
-            {"Amount": "52.92", "Rate": "15.0", "Type": False},
-            {"Amount": "105.84", "Rate": "30.0", "Type": False},
+            {"Type": False, "Rate": "15.0", "Amount": "52.92"},
```
Comment (pedrobaeza) [inline edi_voxel_account_invoice_oca/tests/test_voxel_account_invoice.py]: Isn't the dict comparison order non-sensitive?

File: edi_voxel_account_invoice_oca/tests/test_voxel_account_invoice.py (hunk 3 above)
Comment (FernandoRomera) [reply edi_voxel_account_invoice_oca/tests/test_voxel_account_invoice.py]: Add another default value, field "partner_shipping_id": False,

File: edi_voxel_account_invoice_oca/tests/test_voxel_account_invoice.py (hunk 4 above)
Comment (FernandoRomera) [reply edi_voxel_account_invoice_oca/tests/test_voxel_account_invoice.py]: Yes, plugin VSCode, Restore.

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump
